### ISSUE TITLE
ArgumentError: Call `select' with at least one field

### ISSUE THREAD

**Issue by dlcmh** [shioyama/mobility#358]
Happens when using Ruby's `Array#select` with `ActiveRecord_Relation` & `ActiveRecord_Associations_CollectionProxy` instances.

```ruby
parents = Parent.employed
parents.select { |r| some_algo(r.salary) == :great_employee }

# Exception: ArgumentError: Call `select' with at least one field
# --
# 0: /Users/dlcmh/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/activerecord-6.0.2.1/lib/active_record/relation/query_methods.rb:274:in `select'
# 1: /Users/dlcmh/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/mobility-0.8.9/lib/mobility/plugins/active_record/query.rb:143:in `block (2 levels) in <module:QueryExtension>'
```

I was able to get it working by introducing `&block` & a single line to https://github.com/shioyama/mobility/blob/v0.8.9/lib/mobility/plugins/active_record/query.rb#L140:

```ruby
if ::ActiveRecord::VERSION::STRING >= '5.0'
  %w[pluck group select].each do |method_name|
    define_method method_name do |*attrs, &block| # Add &block
      return super(&block) if (method_name == 'select' && block.present?) # Add this line
```

Any thoughts on whether this will constitute a valid patch? Or will stuff break elsewhere?

Thanks!

**Comment by shioyama** [comment]
Sorry missed this one, could you make a PR with your proposed change? Thanks 🙇‍♂️

**Comment by thealiilman** [comment]
Hey @shioyama, no worries. 🤓 I don't know if you have noticed it, but @dlcmh created a PR already. #359

**Comment by thealiilman** [comment]
Closed by https://github.com/shioyama/mobility/pull/359 and Chris.

**Comment by shioyama** [comment]
This has been resolved by #359, closing.
